sentry-kotlin-multiplatform/src/appleMain/kotlin/io/sentry/kotlin/multiplatform/ScopeCocoaImpl.kt

```diff
@@ -0,0 +1,126 @@
+package io.sentry.kotlin.multiplatform
+
+import io.sentry.kotlin.multiplatform.extensions.*
+import io.sentry.kotlin.multiplatform.protocol.Breadcrumb
+import io.sentry.kotlin.multiplatform.protocol.User
+import platform.Foundation.NSDictionary
+import platform.Foundation.allKeys
+
+internal class ScopeCocoaImpl(private val scope: CocoaScope) : ISentryScope {
+
+    override var level: SentryLevel?
+        set(value) {
+            value?.let { scope.setLevel(it.toCocoaSentryLevel()) }
+        }
+        get() {
+            val level = scope.serialize()["level"]
+            return level?.let { SentryLevel.valueOf((it as String).uppercase()) }
+        }
+
+    override var user: User?
+        set(value) {
+            scope.setUser(value?.toCocoaUser())
+        }
+        get() {
+            val map = scope.serialize()["user"] as? Map<String, String>?
+            return map?.let { User.fromMap(map) }
```

Comment:
m: This is probably so slow.
Can't we get the user property directly?

Reply:
Cocoa sdk doesn't have public getters for the properties in scope, only `setUser()`. I'll open a feature request for that soon

Reply:
I dont think we want to add this to cocoa SDK. But, we do have a private header (is like an internal thing) with this and other properties. 
Can KVM add this private headers in the cocoa bridge?

Reply:
hm good point, yes I think there's a possibility to create a bridge to get those, I'll investigate that

Reply:
@brustolin I added custom headers to access the properties, no need for serialize now